src/shared/json-queue.c

```diff
@@ -105,8 +133,260 @@ cJSON * jqueue_next(file_queue * queue) {
     }
 }
 
+/* Re Handle the file queue */
+static int Handle_JQueue(file_queue *fileq, int flags) {
+    /* Close if it is open */
+    if (!(flags & CRALERT_FP_SET)) {
+        if (fileq->fp) {
+            fclose(fileq->fp);
+            fileq->fp = NULL;
+        }
+
+        /*
+            We must be able to open the file, fseek and get the
+            time of change from it.
+        */
+        fileq->fp = fopen(fileq->file_name, "r");
+        if (!fileq->fp) {
+            /* Queue not available */
+            return 0;
+        }
+    }
+
+    /* Seek to the end of the file */
+    if (!(flags & CRALERT_READ_ALL)) {
+        if (!fileq->fp) {
+            return 0;
+        }
+
+        if (fseek(fileq->fp, 0, SEEK_END) < 0) {
+            merror(FSEEK_ERROR, fileq->file_name, errno, strerror(errno));
+            fclose(fileq->fp);
+            fileq->fp = NULL;
+            return -1;
+        }
+    }
+
+    /* File change time */
+    if (fileq->fp) {
+        if (fstat(fileno(fileq->fp), &fileq->f_status) < 0) {
+            merror(FSTAT_ERROR, fileq->file_name, errno, strerror(errno));
+            fclose(fileq->fp);
+            fileq->fp = NULL;
+            return -1;
+        }
+    }
+
+    fileq->last_change = fileq->f_status.st_mtime;
+
+    return 1;
+}
+
 // Close queue
 void jqueue_close(file_queue * queue) {
     fclose(queue->fp);
     queue->fp = NULL;
 }
+
+/* Set queue flags */
+void jqueue_flags(file_queue *fileq, int flags) {
+    fileq->flags = flags;
+}
+
+/* Return alert data for the next file in the queue */
+alert_data *GetAlertJSONData(file_queue *fileq) {
+    alert_data *al_data;
+    cJSON *al_json;
+    char *groups;
+    int i = 0;
+
+    cJSON *json_object;
+    cJSON *timestamp;
+    cJSON *rule;
+    cJSON *syscheck;
+    cJSON *location;
+    cJSON *srcip;
+    cJSON *full_log;
+
+    os_calloc(1, sizeof(alert_data), al_data);
+
+    /* Get message if available */
+    al_json = jqueue_next(fileq);
+
+    if (!al_json || !fileq->fp) {
+        cJSON_Delete(al_json);
+        FreeAlertData(al_data);
+        return NULL;
+    }
+
+    /* Date */
+    timestamp = cJSON_GetObjectItem(al_json, "timestamp");
+
+    if (timestamp) {
+        os_strdup(timestamp->valuestring, al_data->date);
+    }
+
+    /* Rule */
+    rule = cJSON_GetObjectItem(al_json, "rule");
+
+    if (!rule) {
+        cJSON_Delete(al_json);
```

Comment:
Please, add an error/warning message to know that it received a malformed alert without the `rule` field.

Reply:
Error message added in line: https://github.com/wazuh/wazuh/pull/4256/files#diff-4389a0705fc67fca0285ad7bf1614e46R235